scripts/quests/ahtUrhgan/Promotion_Lance_Corporal.lua

```diff
@@ -0,0 +1,252 @@
+                            if not player:hasKeyItem(tube) then
+                                progress = false
+                            end
+                        end
+
+                        if progress then
+                            quest:setVar(player, 'Prog', 2)
+                        end
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and
+                vars.Prog >= 2
+        end,
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Nafiwaa'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 2 then
+                        return quest:progressEvent(5038, { text_table = 0, })
+                    end
+                end,
+            },
+
+            ['Abquhbah'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        return quest:event(5034, { text_table = 0 })
+                    else
+                        return quest:event(5033, { text_table = 0 })
+                    end
+                end,
+            },
+
+            ['Naja_Salaheem'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        quest:getVar(player, 'Stage') <= VanadielUniqueDay() and
+                        quest:getVar(player, 'Prog') == 3 and
+                        quest:getMustZone(player)
+                    then
+                        return quest:progressEvent(5031, { text_table = 0 })
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [5031] = function(player, csid, option, npc)
+                    local score = quest:getVar(player, 'Score')
+
+                    if score == 1 then
+                        quest.reward.item = xi.items.IMPERIAL_GOLD_PIECE
+                    elseif score == 2 then
+                        quest.reward.item = { xi.items.IMPERIAL_GOLD_PIECE, 2 }
+                    end
+
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.SP_WILDCAT_BADGE)
+                        player:setCharVar("AssaultPromotion", 0)
+                        player:messageSpecial(ID.text.LC_PROMOTION)
+                    end
+                end,
+
+                [5038] = function(player, csid, option, npc)
+                    -- Throwaway: Option == 1073741839
+                    if option ~= 1073741839 then
+                        quest:setVar(player, 'Stage', VanadielUniqueDay())
+                        quest:setVar(player, 'Prog', 3)
+                        quest:setMustZone(player)
+
+                        if option == 2147483990 then -- Platinum
```

Comment:
we have mapped out 250 combinations that result in platinum and 6 combos for Luminium

Reply:
you guys put in all that work?! :0 I'm impressed.